src/channeldef.ts

```diff
@@ -943,11 +943,11 @@ export function title(
   }
 }
 
-export function getGuide(fieldDef: TypedFieldDef<string> | SecondaryFieldDef<string> | DatumDef): Guide {
+export function getGuide(fieldDef: TypedFieldDef<string> | SecondaryFieldDef<string> | DatumDef): Guide<SignalRef> {
   if (isPositionFieldOrDatumDef(fieldDef) && fieldDef.axis) {
-    return fieldDef.axis;
+    return replaceExprRef(fieldDef.axis);
```

Comment:
@kanitw is this the right place to replace the expressway ref?

Reply:
I searched a bit and replaceExprRef seems to happen consistently during `init()`.

A related problem is that this getGuide is now like anti-pattern because it was written before we make expr/signal generic type of axis/legend (but not FieldDef).

Reply:
I'm not following. What is the anti pattern here?